### PR TITLE
Add _git module

### PR DESCRIPTION
**Pull request by saicheems**
Contains convenience functions and classes for cloning and operating on
Git repositories.

### CODE REVIEW

File: server/tasks/_git.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Implements a utility for cloning and operating on Git repositories."""
+import re
+from enum import Enum
+
+from tasks._check_output import check_output
+
+_DIFF_NAME_STATUS_RE = re.compile(r'^(\w)\s+(.*)$')
```
Comment (vchudnov-g) [inline server/tasks/_git.py]: comment

File: server/tasks/_git.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            list(str): a list of emails of the authors of all commits since
+                `rev`.
+        """
+        data = check_output(
+            ['git', 'log', '{}..HEAD'.format(rev), '--pretty=format:"%ae"'],
+            cwd=self.filepath)
+        return data.strip().split('\n')
+
+    def checkout(self, branch):
+        """Switch branches.
```
Comment (vchudnov-g) [inline server/tasks/_git.py]: "Switches"

File: server/tasks/_git.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        Args:
+            branch (str): the name of the branch to checkout.
+
+        Raises:
+            CallError: if the call returns a non-zero return code.
+        """
+        check_output(['git', 'checkout', branch], cwd=self.filepath)
+
+    def commit(self, message, name, email):
+        """Record changes to the repository.
```
Comment (vchudnov-g) [inline server/tasks/_git.py]: "Records"

File: server/tasks/_git.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        """
+        check_output(
+            ['git',
+             '-c', 'user.name={}'.format(name),
+             '-c', 'user.email={}'.format(email),
+             'commit', '-a', '--allow-empty-message', '-m', message],
+            cwd=self.filepath)
+
+    def diff_name_status(self, rev=None, staged=True):
+        """Return a list of status, filename pairs of changes from `rev`.
```
Comment (vchudnov-g) [inline server/tasks/_git.py]: "Returns"

**Comment by saicheems** [comment]
So I spoke with Ethan some time ago about making parts of artman and dartman compatible... and I suggested that we refactor in the future, once this server is at 100%.

I think if it becomes a concern that we repeat a lot of common code (I don't think it should be too big an issue, outside of this git stuff), we should refactor a "googleapis/artman-common" python module that hosts those modules.

Probably not worth the effort right away though.

**Pull request merged**